plugins/orchestrator/src/components/ExecuteWorkflowPage/ExecuteWorkflowPage.tsx

```diff
@@ -150,26 +184,40 @@ export const ExecuteWorkflowPage = (props: ExecuteWorkflowPageProps) => {
             </>
           }
         >
-          {schemaResponse?.schema ? (
+          {schemaResponse?.dataInputSchema?.mainSchema ? (
             <WrappedForm
-              schema={schemaResponse.schema as JSONSchema7}
+              ref={formRef}
+              schema={schemaResponse.dataInputSchema.mainSchema}
               validator={validator}
               showErrorList={false}
-              onSubmit={onExecute}
               onChange={onFormChanged}
               formData={formState}
               formContext={{ formData: formState }}
               templates={{ TitleFieldTemplate }}
+              liveValidate={liveFormValidate}
+              noHtml5Validate
             >
               {executeButton}
             </WrappedForm>
           ) : (
             <Grid container spacing={2} direction="column">
               <Grid item>
```

Comment:
Is there a case where no properties are needed? In this case do we want to provide this text area?

Reply:
Yes, there could be a case where no properties are needed but we are not able to know it at runtime. This is up to the workflow definition developer. There could also be a case where input parameters are needed but no input schema was defined, that's why we are providing an optional "backup" mechanism to pass inputs.

We have something similar in one of our apps.
<img width="500" alt="image" src="https://github.com/caponetto/backstage-plugins/assets/638737/5e8b0f6f-5a46-40a3-997b-b8dd98c80c71">